pygmt/datasets/samples.py

```diff
@@ -20,6 +20,19 @@ def list_sample_data():
     See Also
     --------
     load_sample_data : Load an example dataset from the GMT server.
+
+
+    Examples
+    --------
+    >>> import pygmt
+    >>> pygmt.datasets.list_sample_data()
+    {'bathymetry': 'Table of ship bathymetric observations off Baja California',
+    'fractures': 'Table of hypothetical fracture lengths and azimuths',
+    'hotspots': 'Table of locations, names, and symbol sizes of hotpots from  Mueller et al., 1993',
+    'japan_quakes': 'Table of earthquakes around Japan from NOAA NGDC database',
+    'mars_shape': 'Table of topographic signature of the hemispheric dichotomy of  Mars from Smith and Zuber (1996)',
+    'ocean_ridge_points': 'Table of ocean ridge points for the entire world',
+    'usgs_quakes': 'Table of global earthquakes from the USGS'}
```

Comment:
Should this list only 2-3 dataset names, and maybe use `...` to indicate that this is an incomplete list? I don't want users to think the output in the example is the complete list.
```
   {'bathymetry': 'Table of ship bathymetric observations off Baja California',
    'fractures': 'Table of hypothetical fracture lengths and azimuths',
   ...
    'usgs_quakes': 'Table of global earthquakes from the USGS'}
```

Reply:
See https://github.com/GenericMappingTools/pygmt/issues/1774. The idea of this PR is to better document the available sample data.

Reply:
Are you proposing deprecating list_sample_data?

Reply:
No, `list_sample_data` is still useful to show the available datasets in interactive presentations.